jacqueline-chamnankool/wk01/2-tues/strings/strings.js

```diff
@@ -0,0 +1,129 @@
+// https://www.teaching-materials.org/javascript/exercises/strings
+
+/* DrEvil
+Create a function called DrEvil. It should take a single argument, an amount, 
+and return '<amount> dollars', except it will add '(pinky)' at the end if the 
+amount is 1 million. */
+function DrEvil(amount) {
+    let str = amount + " dollars";
+    if (amount == 1000000) {
+        str += " (pinky)";
+    }
+
+    return str;
+}
+
+/* MixUp
+Create a function called mixUp. It should take in two strings, and return 
+the concatenation of the two strings (separated by a space) slicing out and 
+swapping the first 2 characters of each. You can assume that the strings 
+are at least 2 characters long. */
+function mixUp(str1, str2) {
+    let tempChar = str1[0];
+    str1 = str1.replace(tempChar, str2[0]);
+    str2 = str2.replace(str2[0], tempChar);
+    //string.replace(character to replace, character to be replaced)
+    return str1 + " " + str2;
+}
+
+/* FixStart
+Create a function called fixStart. 
+It should take a single argument, a string, and return a version where all 
+occurences of its first character have been replaced with '*', except for the 
+first character itself. You can assume that the string is at least one 
+character long. */
+function fixStart(str) {
+    //need global g modifier
+    let firstChar = str[0];
+    //allows using variable in replace function
+    var reg = new RegExp(firstChar, "g");
+    let result = firstChar + str.slice(1).replace(reg, "*");
+    return result;
+}
+
+/* Verbing 
+Create a function called verbing. It should take a single argument, a string. 
+If its length is at least 3, it should add 'ing' to its end, unless it already 
+ends in 'ing', in which case it should add 'ly' instead. If the string length 
+is less than 3, it should leave it unchanged. */
+function verbing(str) {
+    let ending = "";
+    //get last 3 letters
+    for (let i = str.length-3; i < str.length; i++) {
+        ending += str[i];
+    }
+
+    if (str.length >= 3 && ending !== "ing") {
+        str += "ing";
+    } else if (str.length >= 3 && ending == "ing") {
+        str += "ly";
+    }
```

Comment:
```suggestion
    if (str.length >= 3) {
       if (ending !== "ing") {
           str += "ing";
        } else {
           str += "ly";
        }
    }
```
_Tip: if you nest if-else statements like this then you may be able to avoid repeating your condition expressions (such as `str.length >= 3`)._

We'll be discussing the concept of DRY code later in class and will encourage you to DRY up your code more and more as the course progresses.